tests/codegen/queries/optional_list_of_people.graphql

```diff
@@ -0,0 +1,6 @@
+query OptionalListOfPeople {
```

Comment:
Let's also rename this file (and the corresponding snapshot files) to better describe the test case. All other files in this folder are named after what they are testing in GraphQL terms. From the current name, it might not be clear why we need to test an "optional list of people." The file could be named `nullable_list_of_non_scalars.graphql` instead, matching your PR's description.